rskj-core/src/main/java/org/ethereum/vm/program/invoke/ProgramInvokeFactoryImpl.java

```diff
@@ -61,7 +62,7 @@ public ProgramInvoke createProgramInvoke(Transaction tx, int txindex, Block bloc
         byte[] caller = tx.getSender(signatureCache).getBytes();
 
         /***         BALANCE op       ***/
-        Coin balance = repository.getBalance(addr);
+        Coin balance = Optional.ofNullable(repository.getBalance(addr)).orElse(Coin.ZERO);
 
```

Comment:
wondering why do we need this change?

Reply:
This is because of the refactor done in `TransactionExecutor` to create the `programInvoke` variable outsite of the [else block](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L354) it originally was. After the refactor the `repository.getBalance(addr)` returns null if the current [call()](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L295) method execution was not meant to enter that [else block](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L354).